src/layout/layout.rs

```diff
@@ -436,333 +443,267 @@ impl Layout {
-                .take(elements.len().saturating_add(1)) // one more than number of elements
-                .try_collect()?;
-                // spacers growing should be the lowest priority
-                for spacer in spacers.iter() {
-                    solver.add_constraint(spacer.size() | EQ(WEAK) | area_size)?;
+            Flex::SpaceBetween => {
+                for (left, right) in spacers.iter().skip(1).rev().skip(1).tuple_combinations() {
+                    solver.add_constraint(left.equals_size_of(right, strengths::SPACER_SIZE_EQ))?
                 }
-                // Spacers should all be similar in size
-                // these constraints should not be stronger than existing constraints
-                // but if they are weaker `Min` and `Max` won't be pushed to their desired values
-                // I found using `STRONG` gives the most desirable behavior
-                for (left, right) in spacers.iter().tuple_combinations() {
-                    solver.add_constraint(left.size() | EQ(STRONG) | right.size())?;
+                for spacer in spacers.iter().skip(1).rev().skip(1) {
+                    solver.add_constraint(spacer.equals_size_to(size, strengths::SPACE_GROWER))?;
                 }
-                // interleave spacers and elements
-                // for `SpaceAround` we want the following
-                // `[spacer, element, spacer, element, ..., element, spacer]`
-                // this is why we use one more spacer than elements
-                for pair in Itertools::interleave(spacers.iter(), elements.iter())
-                    .collect::<Vec<&Element>>()
-                    .windows(2)
-                {
-                    solver.add_constraint(pair[0].end | EQ(REQUIRED) | pair[1].start)?;
+                if let Some(first_segment) = segments.first() {
+                    solver.add_constraint(first_segment.start_sticks_to(start))?
+                }
+                if let Some(last_segment) = segments.last() {
+                    solver.add_constraint(last_segment.end_sticks_to(end))?
                 }
+                // solver.add_constraint(segments.first().map(|e| e.stick_start_to(start)).ok_or(
+                //     AddConstraintError::InternalSolverError("Unable to get first segment"),
+                // )?)?;
+                // solver.add_constraint(segments.last().map(|e| e.stick_end_to(end)).ok_or(
+                //     AddConstraintError::InternalSolverError("Unable to get last segment"),
+                // )?)?;
             }
             Flex::StretchLast => {
-                // this is the default behavior
-                // by default cassowary tends to put excess into the last constraint of the lowest
-                // priority.
-                if let Some(first) = elements.first() {
-                    solver.add_constraint(first.start | EQ(REQUIRED) | area_start)?;
+                for spacer in spacers.iter().skip(1).rev().skip(1) {
```

Comment:
What's the intent of this? It seems a bit magical (opaque in operation).

Reply:
We have to skip the first and last spacers when adding constraints for everything except `SpaceAround`. Would a comment suffice in clarifying intent?

Reply:
yep, something like `//ignore first and last spacer` would do it, and perhaps combine with `spacers[1..spacers.len-1]` as this is a slice.

Reply:
note - I just realized why that doesn't work :)
https://docs.rs/itertools/0.11.0/itertools/trait.Itertools.html#method.with_position is an approach that also could help:
```rust
spacers.with_position().filter(|(_,position) position = Position::Middle).map((e,_) e)
```
Maybe it's just as complex?

Reply:
I added a comment. `skip(1).rev().skip(1)` seems as clear to me as `with_position`, so I'm indifferent.